src/NuGet.Clients/NuGet.PackageManagement.UI/Models/LicenseFileText.cs

```diff
@@ -14,29 +17,36 @@ internal class LicenseFileText : IText, INotifyPropertyChanged
         private string _text;
         private FlowDocument _licenseText;
         private string _licenseHeader;
+        private string _packagePath;
         private readonly string _licenseFileLocation;
-        private Func<string, string> _loadFileFromPackage;
+        private PackageIdentity _packageIdentity;
 
         private int _initialized;
 
-        internal LicenseFileText(string text, string licenseFileHeader, Func<string, string> loadFileFromPackage, string licenseFileLocation)
+        internal LicenseFileText(string text, string licenseFileHeader, string packagePath, string licenseFileLocation, PackageIdentity packageIdentity)
         {
             _text = text;
             _licenseHeader = licenseFileHeader;
             _licenseText = new FlowDocument(new Paragraph(new Run(Resources.LicenseFile_Loading)));
-            _loadFileFromPackage = loadFileFromPackage;
+            _packagePath = packagePath;
             _licenseFileLocation = licenseFileLocation;
+            _packageIdentity = packageIdentity;
         }
 
         internal void LoadLicenseFile()
         {
             if (Interlocked.CompareExchange(ref _initialized, 1, 0) == 0)
             {
-                if (_loadFileFromPackage != null)
+                if (_packagePath != null)
                 {
                     NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                     {
-                        var content = _loadFileFromPackage(_licenseFileLocation);
+                        IServiceBrokerProvider serviceBrokerProvider = await ServiceLocator.GetInstanceAsync<IServiceBrokerProvider>();
+                        IServiceBroker serviceBroker = await serviceBrokerProvider.GetAsync();
+
+                        var embeddedFileUri = new Uri(_packagePath + "#" + _licenseFileLocation);
```

Comment:
am I missing something? Where is this used? Did you intend to pass this as the first argument to GetEmbeddedLicenseAsync?

Reply:
good catch... the API used to take the URI...i forgot to clean this up. fixed in next iteration.